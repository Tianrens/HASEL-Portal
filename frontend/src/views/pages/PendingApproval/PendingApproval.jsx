import React from 'react';
import TextField from '@mui/material/TextField';
import { container, header, title, value, wrapper } from './PendingApproval.module.scss';
import StyledHeader from '../../components/text/StyledHeader';
import TopBarPageTemplate from '../../components/templates/TopBarPageTemplate/TopBarPageTemplate';
import { useDoc } from '../../../state/state';
import { userDoc } from '../../../state/docs/userDoc';

const PendingApproval = () => {
    const [user] = useDoc(userDoc);
    const request = user.currentRequestId;

    const supervisorName = request?.supervisorName ?? '--';
    const comments = request?.comments ?? '--';
    const workstationName = request?.allocatedWorkstationId?.name ?? '--';

    return (
        <TopBarPageTemplate>
            <div className={wrapper}>
                <div className={container}>
                    <h1 className={header}>Sit tight, your account is awaiting approval.</h1>
                    <StyledHeader>Your Application</StyledHeader>
                    <h3 className={value}>
                        <span className={title}>Supervisor Name: </span>
                        {supervisorName}
                    </h3>
                    <h3 className={title}>Reasoning/Comments:</h3>
                    <TextField
                        value={comments}
                        multiline
                        rows={5}
                        InputProps={{
                            readOnly: true,
                            style: { fontSize: '1.17rem', lineHeight: '1.8rem' },
                        }}
                        variant='outlined'
                    />
                    <h3 className={value}>
                        <span className={title}>Workstation Requested: </span>
                        {workstationName}
                    </h3>
                </div>
            </div>
        </TopBarPageTemplate>
    );
};

export default PendingApproval;
