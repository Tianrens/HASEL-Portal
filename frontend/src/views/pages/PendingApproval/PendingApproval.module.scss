@import '../../../assets/colours';

.header {
    color: $heading;
    text-align: center;
    margin-bottom: 4rem;
}
.value {
    font-weight: normal;
    margin-top: 2rem;
    margin-bottom: 0.5rem;
}

.container {
    display: flex;
    flex-direction: column;
    width: 60%;
    height: 100%;
    margin: auto;
}

.title {
    margin-top: 2rem;
    margin-bottom: 0.5rem;
    color: $secondaryBlue;
    font-weight: bold;
}

.reasonField {
    pointer-events: none;
}
