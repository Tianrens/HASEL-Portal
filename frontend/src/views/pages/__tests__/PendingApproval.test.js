import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import PendingApproval from '../PendingApproval/PendingApproval';

test('Pending Approval Page renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <PendingApproval />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
