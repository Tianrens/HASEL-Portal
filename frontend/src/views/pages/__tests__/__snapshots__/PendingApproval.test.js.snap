// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pending Approval Page renders properly 1`] = `
Array [
  <div
    className="fillTopbarSpace"
  >
    <div
      className="navbar"
    >
      <div
        className="patternBgLayer"
      >
        <div
          className="bgLayer"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="University of Auckland"
              className="uoaLogo"
              src="uoaLogo.svg"
            />
            <div
              className="spacer"
            />
            <div
              className="title"
            >
              HASEL Lab
            </div>
          </a>
          <div
            className="pushSpacer"
          />
          <div
            className="horizontalNavLinks"
          >
            <div
              className="linksWrapper"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text wrapper"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <a
                    href="/user"
                    onClick={[Function]}
                  >
                    <img
                      alt="My Account"
                      className="navIcon"
                      src="id.svg"
                    />
                    <div
                      className="spacer"
                    />
                    <div
                      className="title"
                    >
                      My Account
                    </div>
                  </a>
                </span>
              </button>
            </div>
          </div>
          <div
            className="drawerWrapper"
            onClick={[Function]}
            onTouchEnd={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root drawerButton"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2 15.5v2h20v-2H2zm0-5v2h20v-2H2zm0-5v2h20v-2H2z"
                  />
                </svg>
              </span>
            </button>
            <div
              className="MuiDrawer-root MuiDrawer-docked drawer"
            >
              <div
                className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
                style={
                  Object {
                    "visibility": "hidden",
                  }
                }
              >
                <div
                  className="drawerPatternBg"
                >
                  <div
                    className="drawer"
                  >
                    <div>
                      <button
                        className="MuiButtonBase-root MuiIconButton-root drawerButton"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    <div
                      className="linksWrapper"
                    >
                      <button
                        className="MuiButtonBase-root MuiButton-root MuiButton-text wrapper"
                        disabled={false}
                        onBlur={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiButton-label"
                        >
                          <a
                            href="/user"
                            onClick={[Function]}
                          >
                            <img
                              alt="My Account"
                              className="navIcon"
                              src="id.svg"
                            />
                            <div
                              className="spacer"
                            />
                            <div
                              className="title"
                            >
                              My Account
                            </div>
                          </a>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="content"
  >
    <div
      className="container"
    >
      <h1
        className="header"
      >
        Sit tight, your account is awaiting approval.
      </h1>
      <div
        className="container"
      >
        <h2
          className="text "
        >
          <span>
            Your Application
          </span>
        </h2>
      </div>
      <h3
        className="value"
      >
        <span
          className="title"
        >
          Supervisor Name: 
        </span>
        --
      </h3>
      <h3
        className="title"
      >
        Reasoning/Comments:
      </h3>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
          onClick={[Function]}
          style={
            Object {
              "fontSize": "1.17rem",
              "lineHeight": "1.8rem",
            }
          }
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            rows={5}
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-2"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <h3
        className="value"
      >
        <span
          className="title"
        >
          Workstation Requested: 
        </span>
        --
      </h3>
    </div>
  </div>,
]
`;
